infra/main.bicep

```diff
@@ -418,11 +418,8 @@ module searchService 'core/search/search-services.bicep' = {
     name: !empty(searchServiceName) ? searchServiceName : 'gptkb-${resourceToken}'
     location: !empty(searchServiceLocation) ? searchServiceLocation : location
     tags: tags
-    authOptions: {
-      aadOrApiKey: {
-        aadAuthFailureMode: 'http401WithBearerChallenge'
```

Comment:
Was this doing anything for us?? I didn't notice a change with local server, local prepdocs, or deployed app.

Reply:
This explicitly sets that we can use both aad auth and api key auth. If a customer is using a pre-existing service that only supports api key auth they might notice a problem if this template isn't present because it would enable rbac

Reply:
Ah. I don't think we had customers doing that, since our app.py code only fetches a secret if there's a keyvault and a secret name for that keyvault (i.e. the changes we introduced for free tier support).

I could add these authOptions back only for free tier, but are they necessary? Looks like I need to test my free tier environment.

Reply:
My free environment seems to also work fine with this change (local, prepdocs, and up).

Reply:
ok, sounds like this works